merkle/compact/tree.go

```diff
@@ -71,55 +71,59 @@ type GetNodeFunc func(depth int, index int64) ([]byte, error)
 //
 // TODO(pavelkalinnikov): Make GetNodeFunc get all nodes at once.
 func NewTreeWithState(hasher hashers.LogHasher, size int64, f GetNodeFunc, expectedRoot []byte) (*Tree, error) {
-	sizeBits := bits.Len64(uint64(size))
-
-	r := Tree{
-		hasher: hasher,
-		nodes:  make([][]byte, sizeBits),
-		root:   hasher.EmptyRoot(),
-		size:   size,
+	ids := make([]nodeID, 0, bits.OnesCount64(uint64(size)))
```

Comment:
What happens if this is called with size zero? I think it's ok and hasher.EmptyRoot() falls out below?